### PR TITLE
use as plugin

### PR DESCRIPTION
**Pull request by rsbondi**
This is just to get the conversation started, I think using as a plugin would be beneficial, especially if you are running outside of the default, the plugin knows where to find the rpc file, so configuration easier without needing to launch with env and can be just another entry in the lightning config file, also can be managed from the cli/rpc with the `plugin` command.  

This was a bit more than I expected with the logging, but hopefully straight forward.

I could not get the tests to pass, not on a fresh install, but I don't see how the macaroon header is set in the tests, but I am probably missing something.

Also, if this looks like a good idea, additional readme update required

### CODE REVIEW

**Comment by saubyk** [comment]
Hi, thanks for the PR. 
The APIs as plugin is definitely a better approach. I didn't spent enough time exploring that, as we were focused on getting the UI running with the APIs.

Also, regarding the tests, I need to fix them.

Thanks for all the changes, will review.

**Comment by saubyk** [comment]
Hi @rsbondi 
What do I need to pass in `PATH_TO_PLUGIN`, while executing `lightningd`?

**Comment by rsbondi** [comment]
your local repo directory followed by `/plugin.js`

it should be executable when you check it out? if not chmod it

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   "dependencies": {
     "atob": "^2.1.2",
     "body-parser": "^1.19.0",
+    "clightningjs": "0.0.2",
```
Comment (saubyk) [inline package.json]: May need to fork this library before adding it to our dependencies. Vary of adding freshly developed libraries, due to security reasons.

Comment (rsbondi) [reply package.json]: it is from a vary active c-lightning collaborator

https://github.com/ElementsProject/lightning/commits?author=darosior

Comment (saubyk) [reply package.json]: Will merge in that case. Thanks.

**Comment by saubyk** [comment]
> your local repo directory followed by `/plugin.js`
> 
> it should be executable when you check it out? if not chmod it

Tested it. Works beautifully. Thanks.

**Comment by saubyk** [comment]
@rsbondi  Tests are fixed now in master. Will add more coverage soon.

**Pull request merged**